api/filter_outputs.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
+	"github.com/pkg/errors"
+)
+
+func (api *API) createFilterOutput(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var req createFilterOutputRequest
+
+	if err := api.ParseRequest(r.Body, &req); err != nil {
```

Comment:
Some of the errors encountered by ParseRequest may not be BadRequest ones but we always return that status.

Reply:
The chances of request failing with io.read or json.Unmarshal, for a reason not related to bad input body is very slim.  As errors.Wrap is presenting the full stack trace, it will pinpoint the cause of error so a single status for every situation won't be bad.

Reply:
I disagree but happy to accept your conclusion